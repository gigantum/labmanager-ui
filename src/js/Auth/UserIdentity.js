//vendor
import {
  graphql,
} from 'react-relay'
//mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const userIdentityQuery = graphql`
  query UserIdentityQuery{
    userIdentity{
      id
      username
      email
      givenName
      familyName
      isSessionValid
    }
  }
`;

const UserIdentity = {
  getUserIdentity: () => {

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(userIdentityQuery(), {}).then((response, error) => {

          if(response){
            resolve(response)
          }else{

            reject(response)
          }
        }).catch((error) =>{

          reject(error)
        })
      }

      fetchData()
    })
  },
  removeUserIdentity: () => {
    RemoveUserIdentityMutation(()=>{
      console.log('removed user')
    })
  }

}

export default UserIdentity
