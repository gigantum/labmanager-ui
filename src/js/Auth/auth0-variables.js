export const AUTH_CONFIG = {
  domain: 'calum.auth0.com',
  clientId: '9wFN5CZA4Z9xa1r8KZIZbevpjHG7bAab',
  callbackUrl: window.location.origin + '/callback'
}
