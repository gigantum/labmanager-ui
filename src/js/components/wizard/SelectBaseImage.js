import React from 'react'
import { QueryRenderer, graphql } from 'react-relay'
import environment from './../../createRelayEnvironment'
import AddEnvironmentComponentMutation from './../../mutations/AddEnvironmentComponentMutation'

const BaseImageQuery = graphql`query SelectBaseImageQuery($first: Int!, $cursor: String){
  availableBaseImages(first: $first, after: $cursor){
    edges{
      node{
        id
        component{
          id
          repository
          namespace
          name
          version
          componentClass
        }
        author{
          id
          name
          email
          username
          organization
        }
        info{
          id
          name
          humanName
          description
          versionMajor
          versionMinor
          tags
          icon
        }
        osClass
        osRelease
        server
        namespace
        tag
        availablePackageManagers
      }
      cursor
    }
    pageInfo{
      endCursor
      hasNextPage
      hasPreviousPage
      startCursor
    }
  }
}`

export default class SelectBaseImage extends React.Component {
  constructor(props){
  	super(props);
  	this.state = {
      'modal_visible': false,
      'name': '',
      'description': '',
      'selectedBaseImage': null,
      'selectedBaseImageId': false
    };
  }

  /*
    click handle
    function(object): takes a base image edge
    sets componest state for selectedBaseImageId and selectedBaseImage
  */
  _selectBaseImage(edge){
    this.setState({'selectedBaseImage': edge})
    this.setState({'selectedBaseImageId': edge.node.id})
  }

  /*
    function()
    gets current selectedBaseImage and passes variables to AddEnvironmentComponentMutation
    callback triggers and modal state is changed to  next window
  */
  _createBaseImage(){
    let component = this.state.selectedBaseImage.node.component;
    AddEnvironmentComponentMutation(
      this.props.labbookName,
      'default',
      component.repository,
      component.namespace,
      component.name,
      component.version,
      "clientMutationId",
      component.componentClass,
      () => {
        this.props.setBaseImage(this.state.selectedBaseImage)
        if(this.props.setComponent){
          this.props.setComponent(this.props.nextWindow)
        }
      }
    )
  }

  render(){

    return(
      <div className="SelectBaseImage">

        <p> Base Image </p>

        <QueryRenderer
          variables={{
            first: 20
          }}
          query={BaseImageQuery}
          environment={environment}
          render={({error, props}) =>{

              if(error){

                return(<div>{error.message}</div>)
              }else{

                if(props){
                  return(
                    <div className="SelectBaseImage__inner-container flex flex-column justify--space-between">
                      <div className="SelectBaseImage__selected-image-container">

                          {
                            (this.state.selectedBaseImage !== null) && (
                              <div
                                className="SelectBaseImage__selected-image">
                                <img
                                  alt=""
                                  src={this.state.selectedBaseImage.node.info.icon}
                                  height="50"
                                  width="50"
                                />
                                <p>{this.state.selectedBaseImage.node.info.humanName}</p>
                              </div>
                            )
                          }
                      </div>

                      <div className="SelectBaseImage__images flex flex--row flex--wrap justify--space-around">
                      {
                        props.availableBaseImages.edges.map((edge) => {

                            return(
                              <div
                                className={(this.state.selectedBaseImageId === edge.node.id) ? 'SelectBaseImage__image--selected': 'SelectBaseImage__image'}
                                onClick={()=> this._selectBaseImage(edge)}
                                key={edge.node.id}>
                                  <img
                                    alt=""
                                    src={edge.node.info.icon}
                                    height="50"
                                    width="50"
                                  />
                                  <p>{edge.node.info.humanName}</p>
                              </div>
                            )
                        })
                      }

                      </div>
                    <div className="SelectBaseImage__progress-buttons flex flex--row justify--space-between">
                      <button className="SelectBaseImage__progress-button flat--button">
                        Previous
                      </button>
                      <button className="SelectBaseImage__progress-button flat--button">
                        Cancel
                      </button>
                      <button className="SelectBaseImage__progress-button flat--button">
                        skip
                      </button>
                      <button
                        onClick={()=> this._createBaseImage()} disabled={(!this.state.selectedBaseImageId)}>
                        Save and Continue Setup
                      </button>
                    </div>
                  </div>                  )
                }else{
                  return(<div className="Loading"></div>)
                }
              }
          }}
        />

      </div>
      )
  }
}
