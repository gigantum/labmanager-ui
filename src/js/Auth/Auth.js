import history from 'JS/history';
import auth0 from 'auth0-js';
import { AUTH_CONFIG } from './auth0-variables';
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//store
import store from 'JS/redux/store'


export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience:  AUTH_CONFIG.audience,
    responseType: 'token id_token',
    scope: 'openid profile email user_metadata'
  });

  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.handleAuthentication = this.handleAuthentication.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);

  }

  login() {
    this.auth0.authorize();
  }

  handleAuthentication() {

    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {

        this.setSession(authResult);

      } else if (err) {
        console.error(err);
      //  alert(`Error: ${err.error}. Check the console for further details.`);
      }
    });
  }

  setSession(authResult) {

    // Set the time that the access token will expire at
    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
    localStorage.setItem('access_token', authResult.accessToken);
    localStorage.setItem('id_token', authResult.idToken);
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('family_name', authResult.idTokenPayload.family_name);
    localStorage.setItem('given_name', authResult.idTokenPayload.given_name);
    localStorage.setItem('email', authResult.idTokenPayload.email);
    localStorage.setItem('username', authResult.idTokenPayload.nickname);
    //redirect to labbooks when user logs in

    let storrageRoute = sessionStorage.getItem('CALLBACK_ROUTE')
    let route = storrageRoute !== '' ? storrageRoute : `/labbooks`

    history.replace(route)
  }

  logout() {

    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('family_name')
    localStorage.removeItem('given_name');
    localStorage.removeItem('email');
    localStorage.removeItem('username');
    RemoveUserIdentityMutation(()=>{
      //redirect to root when user logs out
      history.replace('/');
    })
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    let expiresAt = JSON.parse(localStorage.getItem('expires_at'));

    return new Date().getTime() < expiresAt;
  }
}
