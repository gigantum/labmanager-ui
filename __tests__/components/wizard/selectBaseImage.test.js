import React from 'react';
import SelectBaseImage from 'Components/wizard/SelectBaseImage';
import {mount} from 'enzyme'
import renderer from 'react-test-renderer';
import Auth from 'JS/Auth/Auth';
import json from './__relaydata__/SelectBaseImage.json'


const variables = {first:20, labbook: 'demo-lab-book'}
export default variables

let toggleDisabledContinue = () => {}
let setComponent = () => {}
let setLabbookName = () => {}
let setBaseImage = () => {}


test('Test SelectBaseImage rendering', () => {

  const component = renderer.create(


      <SelectBaseImage
        toggleDisabledContinue={toggleDisabledContinue}
        labbookName={'demo-lab-book'}
        setComponent={setComponent}
        setLabbookName={setLabbookName}
        setBaseImage={setBaseImage}
        nextWindow={'selectDevelopmentEnvironment'}
      />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
