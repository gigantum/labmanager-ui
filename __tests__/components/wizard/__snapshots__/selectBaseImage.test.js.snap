// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SelectBaseImage rendering 1`] = `
<div
  className="SelectBaseImage"
>
  <p>
     Base Image 
  </p>
  <div
    className="Loader"
  >
    <svg
      id="circle-middle"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#EDEDED"
        r="6"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
    <svg
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="50"
        cy="50"
        fill="#26A6D1"
        r="4.5"
      />
    </svg>
  </div>
</div>
`;
